server/controllers/userController.ts

```diff
@@ -46,33 +47,65 @@ const createUserController = (
     }
   };
 
+  /**
+   * Fetches a user profile by MongoDB ObjectId.
+   */
+  // const getUserById = async (
+  //   _: Request,
+  //   res: Response,
+  //   next: NextFunction,
+  // ) => {
+  //   try {
+  //     const uid = res.locals.verifiedToken.uid;
+
+  //     logger.debug('Fetching user profile for userId:', uid);
+
+  //     const user = await privateUserService.findUserById(uid);
+
+  //     if (user) {
+  //       res.locals.userProfile = {
+  //         name: user.name,
+  //         profilePicture: user.profilePicture,
+  //         settings: user.settings,
+  //       };
+  //     }
+
+  //     next();
+  //   } catch (error) {
+  //     logger.error('Error fetching user profile for userId', error);
+
+  //     next(new InternalServerError('Failed to fetch user profile'));
+  //   }
+  // };
+
   /**
    * Creates a user profile.
    */
   const createUserProfile = async (
```

Comment:
We probably want to keep the code related to `name` so that when a user signs up by email and password our API can store the provided name.